src/jdk.jfr/share/classes/jdk/jfr/internal/settings/PeriodSetting.java

```diff
@@ -115,7 +115,7 @@ public void setValue(String value) {
         default:
             long nanos = Utils.parseTimespanWithInfinity(value);
             if (nanos != Long.MAX_VALUE) {
-                eventType.setPeriod(nanos / 1_000_000, false, false);
+                eventType.setPeriod(Math.max(1, Math.round(nanos / 1_000_000f)), false, false);
```

Comment:
Is `Math.round()` necessary here? 
We can just keep the integer division which will result to `0` for all arguments < 1_000_000 and the `Math.max()` will take care of providing the correct default.

Another thing to consider - what should happen if the `nanos` value is indeed `0`. Should the event perhaps be disabled?

Reply:
You're correct. I will alter it accordingly.